packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -774,6 +783,109 @@ class AndroidCameraCameraX extends CameraPlatform {
     }
   }
 
+  /// Returns the [ResolutionSelector] that maps to the specified resolution
+  /// preset for camera [UseCase]s.
+  ///
+  /// If the specified [preset] is unavailable, the camera will fall back to the
+  /// closest lower resolution available.
+  ResolutionSelector? _getResolutionSelectorFromPreset(
+      ResolutionPreset? preset) {
+    const int fallbackRule = ResolutionStrategy.fallbackRuleClosestLower;
+
+    Size? boundSize;
+    ResolutionStrategy? resolutionStrategy;
+    switch (preset) {
+      case ResolutionPreset.low:
+        boundSize = const Size(320, 240);
+        break;
+      case ResolutionPreset.medium:
+        boundSize = const Size(720, 480);
+        break;
+      case ResolutionPreset.high:
+        boundSize = const Size(1280, 720);
+        break;
+      case ResolutionPreset.veryHigh:
+        boundSize = const Size(1920, 1080);
+        break;
+      case ResolutionPreset.ultraHigh:
+        boundSize = const Size(3840, 2160);
+        break;
+      case ResolutionPreset.max:
+        // Automatically set strategy to choose highest available.
+        resolutionStrategy = _shouldCreateDetachedObjectForTesting
+            ? ResolutionStrategy.detachedHighestAvailableStrategy()
+            : ResolutionStrategy.highestAvailableStrategy();
+        break;
+      case null:
+        // If not preset is specified, default to CameraX's default behvaior
+        // for each UseCase.
+        return null;
+    }
+
+    if (_shouldCreateDetachedObjectForTesting) {
+      resolutionStrategy ??= ResolutionStrategy.detached(
+          boundSize: boundSize, fallbackRule: fallbackRule);
+      return ResolutionSelector.detached(
+          resolutionStrategy: resolutionStrategy);
+    }
+
+    resolutionStrategy ??=
+        ResolutionStrategy(boundSize: boundSize!, fallbackRule: fallbackRule);
+    return ResolutionSelector(
+        resolutionStrategy: ResolutionStrategy(
+            boundSize: boundSize!, fallbackRule: fallbackRule));
+  }
+
+  /// Returns the [QualitySelector] that maps to the specified resolution
+  /// preset for the camera used only for video capture.
+  ///
+  /// If the specified [preset] is unavailable, the camera will fall back to the
+  /// closest lower resolution available.
+  QualitySelector? _getQualitySelectorFromPreset(ResolutionPreset? preset) {
+    VideoQualityConstraint? videoQuality;
+    switch (preset) {
+      case ResolutionPreset.low:
+      // 240p is not supported by CameraX.
```

Comment:
Alternatively, can specify `VideoQualityConstraint..lowest` here, but we won't know what that is so I defined this as the fallback behavior.